### PR TITLE
Add the total by location and the total by all locations in the tree view in square meters

### PR DESCRIPTION
**Pull request by monroy95**
Modification to tree view to show the area of each location and the total for all locations

location_tree_view_2

### CODE REVIEW

File: erpnext/assets/doctype/location/location.js [hunk 1]
```diff
@@ -11,7 +11,14 @@ frappe.ui.form.on('Location', {
 			};
 		});
 	},
-
+	refresh: function (frm) {
```
Comment (Zlash65) [inline erpnext/assets/doctype/location/location.js]: Why is this needed?
`All Locations` is just a label node displayed on top and each group node under `All Locations` would not have parent node and will be regarded as `root_node` which is kinda wrong. There should only be one root node.

Comment (Zlash65) [inline erpnext/assets/doctype/location/location.js]: Why is this needed?
`All Locations` is just a label node displayed on top and each group node under `All Locations` would not have parent node and will be regarded as `root_node` which is kinda wrong. There should only be one root node.

Comment (Zlash65) [inline erpnext/assets/doctype/location/location.js]: Why is this needed?
`All Locations` is just a label node displayed on top and each group node under `All Locations` would not have parent node and will be regarded as `root_node` which is kinda wrong. There should only be one root node.

Comment (Zlash65) [inline erpnext/assets/doctype/location/location.js]: Why is this needed?
`All Locations` is just a label node displayed on top and each group node under `All Locations` would not have parent node and will be regarded as `root_node` which is kinda wrong. There should only be one root node.

File: erpnext/assets/doctype/location/location_tree.js [hunk 2]
```diff
@@ -27,6 +27,31 @@ frappe.treeview_settings["Location"] = {
 			condition: 'frappe.boot.user.can_create.indexOf("Location") !== -1'
 		}
 	],
+	onrender: function (node) {
```
Comment (Zlash65) [inline erpnext/assets/doctype/location/location_tree.js]: Instead of appending it right after the name, we can make it somewhat like how `Account tree` looks like where the area will be set at the right most side of the label?

Take a look at `account_tree.js` file to see how its done there.

File: erpnext/assets/doctype/location/utils_location.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2018, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+
+@frappe.whitelist()
+def get_total_location():
```
Comment (Zlash65) [inline erpnext/assets/doctype/location/utils_location.py]: This would show wrong result if location filter is applied.
Also no need for new file, we can add this method in `location.py` itself.

File: erpnext/assets/doctype/location/location_tree.js (hunk 2 above)
Comment (Zlash65) [inline erpnext/assets/doctype/location/location_tree.js]: Instead of appending it right after the name, we can make it somewhat like how `Account tree` looks like where the area will be set at the right most side of the label?

Take a look at `account_tree.js` file to see how its done there.

**Review by Zlash65 — CHANGES_REQUESTED** [review]

File: erpnext/assets/doctype/location/utils_location.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    # Get the total of all locations in square meters
+    total_location = frappe.db.sql('''SELECT sum(round(area, 3)) FROM `tabLocation` WHERE is_group = 0''')
```
Comment (Alchez) [inline erpnext/assets/doctype/location/utils_location.py]: You can use the ORM to do this as well.

```python
total_location = frappe.db.get_value("Location", {"is_group": 0}, "sum(round(area, 3))")
```

[the remark this answers is not in the document]
Comment (monroy95) [reply erpnext/assets/doctype/location/utils_location.py]: Great. Thank you.

File: erpnext/assets/doctype/location/location_tree.js (hunk 2 above)
Comment (monroy95) [reply erpnext/assets/doctype/location/location_tree.js]: @Zlash65 New view for location tree: 
location_tree_view

File: erpnext/assets/doctype/location/utils_location.py (hunk 3 above)
Comment (monroy95) [reply erpnext/assets/doctype/location/utils_location.py]: Suggested changes in: e72bd3c

File: erpnext/assets/doctype/location/location.js (hunk 1 above)
Comment (Zlash65) [inline erpnext/assets/doctype/location/location.js]: Why is this needed?
All Locations is just a label node displayed on top and each group node under All Locations would not have parent node and will be regarded as root_node which is kinda wrong. Moreover, `parent_location` isn't a mandatory field either.

File: erpnext/assets/doctype/location/location.py [hunk 5]
```diff
@@ -223,3 +223,10 @@ def add_node():
 
 def on_doctype_update():
 	frappe.db.add_index("Location", ["lft", "rgt"])
+
+@frappe.whitelist()
+def get_total_location():
+	# Get the total of all locations in square meters
+	total_location = frappe.db.get_value("Location", {"is_group": 0}, "sum(round(area, 3))")
```
Comment (Zlash65) [inline erpnext/assets/doctype/location/location.py]: Open the tree view for location and select a location in the filter above. Notice that the tree view will change and the selected filter will become the root of the structure but its area would be wrongly calculated.

**Review by Zlash65 — CHANGES_REQUESTED** [review]

File: erpnext/assets/doctype/location/location.js (hunk 1 above)
Comment (monroy95) [reply erpnext/assets/doctype/location/location.js]: I added it because I understood that each of the nodes groups of all locations are a representation of what they contain and there should be no need to edit the node group and thus not modify the total area per group. Do you suggest to remove it?

Comment (monroy95) [reply erpnext/assets/doctype/location/location.js]: I remain pending the suggestion to push the changes with the solution.

File: erpnext/assets/doctype/location/location.py (hunk 5 above)
Comment (monroy95) [reply erpnext/assets/doctype/location/location.py]: I'm sorry, it's true. I solved it:

solved_tree_view_location

Comment (monroy95) [reply erpnext/assets/doctype/location/location.py]: The suggested changes have been applied to the following  5705af8
